pygments/lexers/usd.py

```diff
@@ -0,0 +1,136 @@
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
+                ),
+                lexer.bygroups(
+                    token.Keyword.Token,
+                    token.Whitespace,
+                    token.Keyword.Token,
+                    token.Whitespace,
+                    token.Keyword.Type,
+                    token.Whitespace,
+                    token.Name.Attribute,
+                    token.Generic,
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+            (
+                r"(custom){_WHITESPACE}{}(\s+){}(\s*)(=)".format(
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
+                ),
+                lexer.bygroups(
+                    token.Keyword.Token,
+                    token.Whitespace,
+                    token.Keyword.Type,
+                    token.Whitespace,
+                    token.Name.Attribute,
+                    token.Generic,
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+            (
+                r"(uniform){_WHITESPACE}{}(\s+){}(\s*)(=)".format(
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
+                ),
+                lexer.bygroups(
+                    token.Keyword.Token,
+                    token.Whitespace,
+                    token.Keyword.Type,
+                    token.Whitespace,
+                    token.Name.Attribute,
+                    token.Generic,
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+            (
+                r"{}{_WHITESPACE}{}(\s*)(=)".format(
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
+                ),
+                lexer.bygroups(
+                    token.Keyword.Type,
+                    token.Whitespace,
+                    token.Name.Attribute,
+                    token.Generic,
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+        ]
+        + _keywords(_usd_builtins.KEYWORDS, token.Keyword.Tokens)
+        + _keywords(_usd_builtins.SPECIAL_NAMES, token.Name.Builtins)
+        + _keywords(_usd_builtins.COMMON_ATTRIBUTES, token.Name.Attribute)
+        + [(r"\b\w+:[\w:]+\b", token.Name.Attribute)]
+        + _keywords(_usd_builtins.OPERATORS, token.Operator)  # more attributes
+        + [(type_ + r"\[\]", token.Keyword.Type) for type_ in _usd_builtins.TYPES]
+        + _keywords(_usd_builtins.TYPES, token.Keyword.Type)
+        + _PUNCTUATION
+        + [
+            ("#.*?$", token.Comment.Single),
+            (",", token.Generic),
```

Comment:
should be either `Text` (no highlighting at all) or `Punctuation`

Reply:
I'll add that for ; too. Looking at it again, the `(r"[\w_:\.]+", Generic)` rule should be Text, as well. Done